### DISCUSSION TITLE
Integrating Markdown Magic with Remark

### DISCUSSION THREAD

**Discussion by ProfessorManhattan** [remarkjs/remark#1138] [Q&A] [answered]
Markdown Magic is similar to Remark in the sense that it helps automate the process of creating templated markdown files. It has some unique features that I would like to leverage. Does anyone know how I can integrate Markdown Magic, similar tools, or even my own code into a shared NPM package configuration for the `remark` CLI so that if I type `remark README.md` it will use all the features?

I'm basically looking for some hook where I can run `markdown-magic` or my own code without having to run two different CLIs on the files.

Is this something that would be best integrated as a plugin?

**Comment by ChristianMurphy** [comment, marked as answer]
Heya @ProfessorManhattan! 👋 

`markdown-magic` works with text and regular expressions, `remark` works on syntax trees and visitor patterns.
The two work in fundamentally different ways.
You could wrap both of them in a shell script or alias which runs both on the same file, back to back if you wanted, the only benefit would be saving a few keystrokes in the terminal.

> It has some unique features that I would like to leverage.

To more efficiently handle files in a single pass.
Consider creating a remark plugin which implements the same feature.
https://unifiedjs.com/learn/ offers some guidance on creating plugins and working with syntax trees.
